src/sentry/replays/consumers/recording/process_recording.py

```diff
@@ -0,0 +1,212 @@
+from __future__ import annotations
+
+import concurrent.futures
+import logging
+from collections import deque
+from concurrent.futures import ALL_COMPLETED, Future, wait
+from io import BytesIO
+from typing import Callable, Deque, Mapping, MutableMapping, NamedTuple, Optional, cast
+
+import msgpack
+import sentry_sdk
+from arroyo import Partition
+from arroyo.backends.kafka.consumer import KafkaPayload
+from arroyo.processing.strategies.abstract import ProcessingStrategy
+from arroyo.types import Message, Position
+from django.conf import settings
+
+from sentry.attachments import MissingAttachmentChunks, attachment_cache
+from sentry.attachments.base import CachedAttachment
+from sentry.models import File
+from sentry.replays.consumers.recording.types import (
+    RecordingSegmentChunkMessage,
+    RecordingSegmentHeaders,
+    RecordingSegmentMessage,
+)
+from sentry.replays.models import ReplayRecordingSegment
+from sentry.utils import json
+
+logger = logging.getLogger("sentry.replays")
+
+CACHE_TIMEOUT = 3600
+
+
+class MissingRecordingSegmentHeaders(ValueError):
+    pass
+
+
+class ReplayRecordingMessageFuture(NamedTuple):
+    """
+    Map a submitted message to a Future returned by the Producer.
+    This is useful for being able to commit the latest offset back
+    to the original consumer.
+    """
+
+    message: Message[KafkaPayload]
+    future: Future[None]
+
+
+class ProcessRecordingSegmentStrategy(ProcessingStrategy[KafkaPayload]):  # type: ignore
+    def __init__(
+        self,
+        commit: Callable[[Mapping[Partition, Position]], None],
+    ) -> None:
+        self.__closed = False
+        self.__futures: Deque[ReplayRecordingMessageFuture] = deque()
+        self.__threadpool = concurrent.futures.ThreadPoolExecutor()
+        self.__commit = commit
+
+    def poll(self) -> None:
+        self._commit_and_prune_futures()
+
+    def _process_chunk(
+        self, message_dict: RecordingSegmentChunkMessage, message: Message[KafkaPayload]
+    ) -> None:
+        # TODO: implement threaded chunk sets, and wait for an individual segment's
+        # futures to finish before trying to read from redis in the final kafka message
+        # https://github.com/getsentry/replay-backend/pull/38/files
+        recording_segment_uuid = message_dict["id"]
+        project_id = message_dict["project_id"]
+        chunk_index = message_dict["chunk_index"]
+        cache_key = replay_recording_segment_cache_id(project_id, recording_segment_uuid)
```

Comment:
This double counts recording_segment_uuid.  The key is formatted `"{key}:a:{id}:{chunk_index}"`.